File: app/api/analysis.py
from fastapi import APIRouter

from app.services.summary_service import SummaryService
from app.services.financial_analysis import FinancialAnalysisService
from app.services.risk_analysis import RiskAnalysisService
from app.services.metrics_service import MetricsService
from app.services.investment_report import InvestmentReportService

router = APIRouter(
    prefix="/analysis",
    tags=["Analysis"],
)

summary_service = SummaryService()
financial_service = FinancialAnalysisService()
risk_service = RiskAnalysisService()
metrics_service = MetricsService()
investment_service = InvestmentReportService()


@router.get("/summary")
def summary():
    result = summary_service.generate_summary()
    return {"markdown": result["answer"]}


@router.get("/financial-analysis")
def financial_analysis():
    result = financial_service.analyze_financials()
    return {"markdown": result["answer"]}


@router.get("/risk-analysis")
def risk_analysis():
    result = risk_service.analyze_risks()
    return {"markdown": result["answer"]}


@router.get("/investment-report")
def investment_report():
    result = investment_service.generate_report()
    return {"markdown": result["answer"]}

File: frontend/src/services/analysis.ts
import { api } from "./api";
import type {
  ExecutiveSummary,
  FinancialAnalysis,
  InvestmentReport,
  MetricsResponse,
  RiskAnalysis,
} from "@/types/analysis";

export async function getExecutiveSummary(): Promise<ExecutiveSummary> {
  const { data } = await api.get<ExecutiveSummary>("/analysis/summary");
  return data;
}

export async function getFinancialAnalysis(): Promise<FinancialAnalysis> {
  const { data } = await api.get<FinancialAnalysis>(
    "/analysis/financial-analysis",
  );
  return data;
}

export async function getRiskAnalysis(): Promise<RiskAnalysis> {
  const { data } = await api.get<RiskAnalysis>("/analysis/risk-analysis");
  return data;
}

export async function getMetrics(): Promise<MetricsResponse> {
  const { data } = await api.get<MetricsResponse>("/analysis/metrics");
  return data;
}

export async function getInvestmentReport(): Promise<InvestmentReport> {
  const { data } = await api.get<InvestmentReport>(
    "/analysis/investment-report",
  );
  return data;
}
